src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java

```diff
@@ -86,7 +86,14 @@ public boolean renameToName(String targetFileName, boolean overwriteExistingFile
         }
 
         final Path oldPath = oldFile.get();
-        final Path newPath = oldPath.resolveSibling(targetFileName);
+        Optional<String> oldExtension = FileUtil.getFileExtension(oldPath);
+        Optional<String> newExtension = FileUtil.getFileExtension(targetFileName);
+
+        if (oldExtension.isEmpty() && newExtension.isEmpty()) {
+            return false;
+        }
```

Comment:
It should still be possible to rename files: E.g., `startUp` to `start-up`. (Even though this is an edge case)